### ISSUE TITLE
Cannot augment types for server in authoring a module, addTypeTemplate does not work for Nitro side

### ISSUE THREAD

**Issue by ymansurozer** [nuxt/nuxt#29695]
### Environment

On Stackblits: 
- Operating System: Linux
- Node Version:     v18.20.3
- Nuxt Version:     3.13.2
- CLI Version:      3.15.0
- Nitro Version:    2.9.7
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      default
- Runtime Modules:  nuxt-auth-utils@0.5.0
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-npuane?file=src%2Fmodule.ts

### Describe the bug

I am authoring a module and have two `d.ts` files where I augment types from Nuxt Auth Utils and Nitro (specifically H3 context) by using `addTypeTemplate`:

`runtime/auth.d.ts`
```ts
declare module '#auth-utils' {
  interface User {
    id: string;
    role: string;
    email: string;
    name: string;
    image?: string;
  }

  interface UserSession {
    user?: User;
  }
}

export {};
```

`runtime/server/h3.d.ts`
```ts
import type { UserSession } from '#auth-utils';

declare module 'h3' {
  interface H3EventContext {
    user?: UserSession['user'];
  }
}
```

`module.ts`
```ts
import {
  defineNuxtModule,
  installModule,
  addTypeTemplate,
  createResolver,
} from '@nuxt/kit';

// Module options TypeScript interface definition
export interface ModuleOptions {}

export default defineNuxtModule<ModuleOptions>({
  meta: {
    name: 'my-module',
    configKey: 'myModule',
  },
  // Default configuration options of the Nuxt module
  defaults: {},
  async setup(_options, _nuxt) {
    await installModule('nuxt-auth-utils');

    const resolver = createResolver(import.meta.url);
    addTypeTemplate({
      filename: 'types/auth.d.ts',
      src: resolver.resolve('runtime/auth.d.ts'),
    });
    addTypeTemplate({
      filename: 'types/h3.d.ts',
      src: resolver.resolve('runtime/server/h3.d.ts'),
    });
  },
});
```

These augmented types work fine in the module's runtime folder. But when I try them in the playground, server-side does not recognize them at all, `event.context` has no `user` object and `getUserSession` returns a `User` but not as I augmented it. Client-side has no issues, `useUserSession` returns the `User` object just as I typed it. 

I am able to verify both files are present in `.nuxt/types` and referenced in `./nuxt/nuxt.d.ts`.

I also tried `addTemplate` and manually adding to the references in `prepare:types` hook as mentioned in the [docs](https://nuxt.com/docs/guide/going-further/modules#adding-type-declarations) but that also did not work.

The only thing that seems to work is to just copy these files to the playground folder, which I really don't want to do.

**Comment by onmax** [comment]
Hey, do you still have the issue?

I am also implementing my own Nuxt Module using utilities from `nuxt-auth-utils`. For the types for the `User` and `Session` I created a `types.d.ts` file next to the `module.ts`:

```ts
declare module '#auth-utils' {
  interface User {
    address: string
    publicKey: string
  }

  interface UserSession {
    challenge: string
  }
}
```

This seems to be working, as if I do `pnpm run dev:prepare` I can see `.nuxt/types/nimiq-auth.d.ts` which is my module.

**Comment by ymansurozer** [comment]
> Hey, do you still have the issue?

Yes, unfortunately. We had to do the workaround I mentioned (which is to copy the `d.ts` file to the app's folder by the module).

**Comment by targetlucked69** [comment]
Have you tried the new version released today @ymansurozer? Doesn't look like it's fixed sadly

**Comment by targetlucked69** [comment]
How does the updated Nuxt with the fix work @danielroe?

I assume if a module author updates their `@nuxt/kit` to latest, and userland updates the module but not Nuxt itself, it will fallback to `compatibilityDate` and augmented types will not work still? Thanks!

**Comment by danielroe** [comment]
augmented types will work as long as you have a version of nuxt/kit that supports this

**Comment by targetlucked69** [comment]
> augmented types will work as long as you have a version of nuxt/kit that supports this

Sorry daniel, still trying to wrap my head around this. Say you're using Nuxt 15 for example, that means you have a `@nuxt/kit` internally of like 3.15-ish and you installed a module that has higher version of `@nuxt/kit`, will your app respect that?

**Comment by danielroe** [comment]
Yes, the logic required is fully in the kit utility, so as long as that is a dependency of your module, you should be good
